File: Spring Boot/lms/src/main/java/com/courzelo/lms/entities/schedule/ElementModule.java
package com.courzelo.lms.entities.schedule;

import com.courzelo.lms.entities.institution.Class;
import com.courzelo.lms.entities.user.User;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.ReadOnlyProperty;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.DocumentReference;

import java.time.DayOfWeek;
import java.util.ArrayList;
import java.util.List;


@Document(collection = "elementModule")
@Getter
@Setter
public class ElementModule {
    @Id
    private  String id;
    @NotNull
    private Integer nmbrHours;
    @NotNull
    @Size(max = 255)
    private String name;
    @NotNull
    private DayOfWeek dayOfWeek;
    @NotNull
    private Period period;
    @DBRef
    private List<Class> classes;
    @NotNull
    @DBRef
    private List<Semester> semesters ;
    @DBRef
    @NotNull
    private List<Department> departments;
    @NotNull
    private int numSemesters;
    @NotNull
    private int numDepartments;
    @DBRef
    private User teacher;
    @DBRef
    private Modul modul;


}

File: AngularFront/src/app/model/schedule/element-module.ts
import {Departement} from "./departement";
import {Semester} from "./semester";
import {Modul} from "./Modul";
import {ClassListDTO} from "../program/ClassListDTO";
import {UserResponse} from "../user/UserResponse";

export interface ElementModule {
id:string;
  nmbrHours: number;
  name: string;
  dayOfWeek:String;
  period: string;
  classes:ClassListDTO;
  semesters: Semester[];
  departments: Departement[];
  numSemesters: number;
  numDepartments: number;
  teacher:UserResponse;
  modul:Modul;

}
